### DISCUSSION TITLE
Deployment Issue. Error: ConvexReactClient requires a URL like 'https://happy-otter-123.convex.cloud'.

### DISCUSSION THREAD

**Discussion by james-randolph-50** [vercel/community#7280] [Help] [answered]
### Summary

I have a NextJS project that uses Convex as the backend. It works locally, but when I go to publish it on Vercel, the build process fails saying: "Error: ConvexReactClient requires a URL like 'https://happy-otter-123.convex.cloud'." 

In my ConvexClientProvider.tsx, I call the convex URL that's in my env.local file, which was generated by convex. I'm able to log the url in my local, so I know the value is being passed.

Has anyone else run into this issue?

### Steps to Reproduce

N/A

**Comment by danusontarangkul** [comment]
Did you ever figure this out?

**Reply by Silver1104** [reply, marked as answer]
You have to create a deployment key separately to get it working. Follow the guide here: https://docs.convex.dev/production/hosting/ 

Choose the deployment you are using. Also, don't forget to change the SITE_URL environmental variable on your convex project from localhost to the domain name.

**Reply by danusontarangkul** [reply]
Thank you!

**Reply by danusontarangkul** [reply]
Hi, I am still getting an error after following the guide

Running "npx convex deploy --cmd 'npm run build'"
✖ Vercel build environment detected but CONVEX_DEPLOY_KEY is not set. Set this environment variable to deploy from this environment. See https://docs.convex.dev/production/hosting

I generated a deployment key from convex and added it as an environmental variable

Please let me know.

<img width="919" alt="Screenshot 2024-09-17 at 5 52 37 PM" src="https://github.com/user-attachments/assets/a3eb1e82-c9ca-4e10-880e-2450558a1423">

**Reply by Silver1104** [reply]
So could you specify where you are encountering the error? Is it being thrown by vercel itself or are you trying to run the program separately

**Reply by danusontarangkul** [reply]
I figured out the issue. I was looking at preview and not deployment. Thank you.

**Reply by Silver1104** [reply]
Great! Glad it worked out.

[a comment by an automated account is not shown]
